README.md

```diff
@@ -119,9 +119,15 @@ If a function or set of functions could reasonably be defined as its own contrac
 
 Note this _does not_ mean that we should avoid inheritance, in general. Inheritance is useful at times, most especially when building on existing, trusted contracts. For example, _do not_ reimplement `Ownable` functionality to avoid inheritance. Inherit `Ownable` from a trusted vendor, such as [OpenZeppelin](https://github.com/OpenZeppelin/openzeppelin-contracts/) or [Solady](https://github.com/Vectorized/solady).
 
-#### 7. Avoid writing interfaces.
+#### 7. Interfaces
 
-Interfaces separate NatSpec from contract logic, requiring readers to do more work to understand the code. For this reason, they should be avoided.
+##### A. Use interfaces over manual abi encodings
+
+When calling external contracts, wrapping an address in an interface to call its functions will provide type-safety and compiler help. Other methods like `abi.encodeWithSelector` or `abi.encodeWithSignature`, while functional, require manual definition without IDE support.
```

Comment:
You don't need an interface for this, right? You can just do `ERC20(address).transfer` are you assuming here there is some pragma conflict?

Reply:
You can just do `Contract(address).function`, but then you bring the entire bytecode of `Contract` into your deployed contract, boosting the size. In extreme cases, you could overflow the 24kb limit without noticing. If you use `IContract`, then you are typically bringing much less bytecode into your contract, saving on deploy costs and also leaving more room for the 24kb limit if needed.

Reply:
Using a contract or interface here is a compiler level thing and doesn't impact bytecode. These will print them exact same thing 
```

contract T is Test {
    function test() public {
        console2.logBytes(type(UseContract).creationCode);
        console2.logBytes(type(UseInterface).creationCode);
    }
}

contract MyContract {
    function call() external {

    }
}

interface MyInterface {
    function call() external;
}

contract UseContract {
    function x() external {
        MyContract(address(1)).call();
    }
}

contract UseInterface {
    function x() external {
        MyInterface(address(1)).call();
    }
}
```

Reply:
It would only be included in bytecode if you did like `new MyContract()`